File: expense-report/python/src/expense_report/exceptions.py
class EmptyReportError(Exception):
    def __init__(self) -> None:
        super().__init__("Cannot submit an empty report")


class ReportExceedsMaximumError(Exception):
    def __init__(self) -> None:
        super().__init__("Report total exceeds the $5,000 maximum")


class InvalidStatusTransitionError(Exception):
    def __init__(self, message: str) -> None:
        super().__init__(message)


class ExpenseNotFoundError(Exception):
    def __init__(self) -> None:
        super().__init__("Expense not found")


class InvalidAmountError(Exception):
    def __init__(self) -> None:
        super().__init__("Amount must be positive")


class FinalizedReportError(Exception):
    def __init__(self) -> None:
        super().__init__("Cannot modify a finalized report")

File: expense-report/typescript/src/Exceptions.ts
export class EmptyReportError extends Error {
  constructor() { super('Cannot submit an empty report'); this.name = 'EmptyReportError'; }
}

export class ReportExceedsMaximumError extends Error {
  constructor() { super('Report total exceeds the $5,000 maximum'); this.name = 'ReportExceedsMaximumError'; }
}

export class InvalidStatusTransitionError extends Error {
  constructor(message: string) { super(message); this.name = 'InvalidStatusTransitionError'; }
}

export class ExpenseNotFoundError extends Error {
  constructor() { super('Expense not found'); this.name = 'ExpenseNotFoundError'; }
}

export class InvalidAmountError extends Error {
  constructor() { super('Amount must be positive'); this.name = 'InvalidAmountError'; }
}

export class FinalizedReportError extends Error {
  constructor() { super('Cannot modify a finalized report'); this.name = 'FinalizedReportError'; }
}
